ramalama/huggingface.py

```diff
@@ -69,6 +67,28 @@ def pull(self, args):
         symlink_dir = os.path.dirname(model_path)
         os.makedirs(symlink_dir, exist_ok=True)
 
+        try:
+            return self.url_pull(args, model_path, directory_path)
+        except (urllib.error.HTTPError, urllib.error.URLError) as e:
```

Comment:
```suggestion
        except (urllib.error.HTTPError, urllib.error.URLError, KeyError) as e:
```
It's actually a KeyError which is then shown to user as 401 Auth err. Adding KeyError here allows it to fallback (for me anyways)